### ISSUE TITLE
In 249 systemd-networkd does not configure .network after systemd-udevd renames .link

### ISSUE THREAD

**Issue by tommy48** [systemd/systemd#20657]
**systemd version the issue has been seen with**
 > 249.2 and 249.4

**Used distribution**
 > gentoo

**Linux kernel version used** (`uname -a`)

 > `Linux deli 5.10.61-gentoo-x86_64 #1 SMP PREEMPT Tue Aug 31 14:31:27 CEST 2021 x86_64 Intel(R) Xeon(R) CPU E5-2420 0 @ 1.90GHz GenuineIntel GNU/Linux`
and `Linux asuz 5.10.61-gentoo-x86_64 #2 SMP PREEMPT Mon Aug 30 13:01:01 CEST 2021 x86_64 Intel(R) Core(TM) i5-7200U CPU @ 2.50GHz GenuineIntel GNU/Linux`

**CPU architecture issue was seen on**
 > …

**Expected behaviour you didn't see**
 > Network on interface lan1 is configured

**Unexpected behaviour you saw**
 > After systemd-networkd "waiting for the interface to be renamed", it never comes back with "lan1: udev initialized link". Nothing more happens, and the interface isn't configured.

**Steps to reproduce the problem**
 > Boot system with simple .link and static .network configs.

```
# cat /etc/systemd/network/10-lan1.link
[Match]
MACAddress=90:b1:1c:10:63:bb

[Link]
Name=lan1
```

```
# cat /etc/systemd/network/30-lan1.network
[Match]
Name=lan1

[Address]
Address=192.168.1.10/24

[Route]
Gateway=192.168.1.1

[Network]
DNS=192.168.1.1
IPv6AcceptRA=no
```

**Additional program output to the terminal or log subsystem illustrating the issue**

```text
…
```

Everything workes fine with 248.6. I've tried with different kernels: 5.10.49, 5.10.52, 5.10.61, but it makes no difference. systemd-248.6 works, systemd-249.2 fails.

I've compared the logs with debug info turned on for networkd and udevd.

The log messages from udevd are the same in both versions.

networkd in 249.2 says "eth0: Saved hardware address: 90:b1:1c:10:63:bb" which 248.6 does not.

Then networkd in 248.6 says some things that are missing in 249.2:
```
eth0: New dvice has no master, continuing without
eth0: State changed: unmanaged -> linger
eth0: Link removed
lan1: New device has no master, continuing without
lan1: Flags change: +MULTICAST +BROADCAST
lan1: Link 2 added
lan1: Interface is being renamed, pending initialization.
lan1: Saved original MTU: 1500
```

**Comment by yuwata** [comment]
Please provide the debugging log of networkd (v249 or newer).

**Comment by tommy48** [comment]
Here's the log. It's for 249.2.  I can do a log for 249.4 if required.

[systemd-networkd.log](https://github.com/systemd/systemd/files/7135880/systemd-networkd.log)

I have a workaround (ExecStartPre=udevadm settle). I understand why that is wrong, but it lets med reboot remotely for now, and I think it suggests either udevd is no longer sending the proper notification or networkd is no longer reacting to it.

**Comment by yuwata** [comment]
Thanks. I think this should be fixed by #20696. If possible, please test it.

**Comment by tommy48** [comment]
I have tested 249.2 and 249.4 patched with commit 160203e, and the network gets configured as expected again.

**Comment by yuwata** [comment]
@tommy48 Thank you for the confirmation.
